test/gpu_tests.jl

```diff
@@ -1246,7 +1246,15 @@ function test_gpu(FT)
         kernel!(output, ip_P3, T, N_l, V_l, Δt; ndrange)
 
         # test if P3_het_N_i is callable and returns reasonable values
-        TT.@test Array(output)[1] ≈ FT(0.0002736160475969029)
+        ref_val = if FT == Float64
+            FT(0.0002736160475969029)
+        else
+            # loss of precision due to
+            # `exp(-B * V_l_converted * Δt * exp(a * Tₛ))` -> 0.9999999 (Float32)
+            # instead of 0.9999998631919762 (Float64).
+            FT(0.00023841858f0)
+        end
+        TT.@test Array(output)[1] ≈ ref_val
```

Comment:
This arises from fixing the type instability [here](https://github.com/CliMA/CloudMicrophysics.jl/pull/621/files#diff-6636c90ace8799ad4bb5048d3240c8d88686a6ac53bea7d01569b609aced7b1bL179-R181)